### DISCUSSION TITLE
Wallet disconnects when page refreshes

### DISCUSSION THREAD

**Discussion by pablohpsilva-forjnetwork** [rainbow-me/rainbowkit#2155] [Q&A]
Since the newest version got released, the `autoConnect` feature was removed from the docs. 
It is really annoying for the users to reconnect every single time they access/refresh the page.

How can I make this work again?

The browsers the user reported are the following:

Edge (latest)
Chrome (latest)
Brave (latest)

**Comment by Adebesin-Cell** [comment]
+1 I have a similar issue, but this time it's with a custom connector. The user's state persists until the page is refreshed. After the page settles, the `wagmi.store` connection state gets cleared. I'm not sure if there's a workaround for this, but it's quite inconvenient if users have to keep reconnecting to use the app.

**Comment by alvinyusuf** [comment]
1. add createStorage in your wagmi config
`export function getConfig() {
  return createConfig({
    chains: [mainnet, sepolia],
    connectors: [
      injected(),
      coinbaseWallet(),
      walletConnect({ projectId: process.env.NEXT_PUBLIC_WC_PROJECT_ID }),
    ],
    storage: createStorage({
      storage: cookieStorage,
    }),
    ssr: true,
    transports: {
      [mainnet.id]: http(),
      [sepolia.id]: http(),
    },
  })
}`
3. add initialState in your WagmiProvider
`    <WagmiProvider config={config} initialState={initialState}>
      <QueryClientProvider client={queryClient}>
        <RainbowKitProvider locale='id-ID'>
          {children}
        </RainbowKitProvider>
      </QueryClientProvider>
    </WagmiProvider>`
5. this code for initialState
` const initialState = cookieToInitialState(
    getConfig(),
    (await headers()).get('cookie'),
  )`
